core/src/program/tests.rs

```diff
@@ -0,0 +1,42 @@
+use super::{Deserializable, Digest, Felt, Kernel, ProgramInfo, Serializable};
+use crate::Word;
+use proptest::prelude::*;
+use rand_utils::prng_array;
+
+proptest! {
+    #[test]
+    fn arbitrary_program_info_serialization_works(
+        kernel_count in prop::num::u8::ANY,
+        ref seed in any::<[u8; 32]>()
+    ) {
+        let mut seed = *seed;
+        let program_hash = digest_from_seed(seed);
+        let kernel: Vec<Digest> = (0..kernel_count)
+            .map(|_| {
+                seed = prng_array(seed);
+                digest_from_seed(seed)
```

Comment:
this is copying the same element over and over, no? maybe `vec![v; len]` is easier to read?

Reply:
The idea is to make every element of the kernel distinct. If we serialize copies of `v`, the serialization implementation might just deserialize it repeatedly and the test case won't intercept such scenario.

For every call of `digest_from_seed`, we expect a different digest as we use the randomized seed as input

Reply:
oh, the code is overwriting the stack state on a map? I didn't notice that. Maybe use `scan` instead?

Reply:
Could be! As we indeed have an internal state of the iterator (the seed), it makes sense. Fixed!